provider/datagen/src/driver.rs

```diff
@@ -147,12 +138,7 @@ impl DatagenDriver {
     // Avoids multiple monomorphizations
```

Comment:
Question: only `sink` gets a trait object; is it worth it?

Reply:
idk, but provider is not object safe so it can't